.travis.yml

```diff
@@ -11,10 +16,10 @@ before_script:
 
 install:
   - pip uninstall -y pytest
-  - pip install -r requirements.txt colorama
+  - pip install -r requirements.txt
 env:
   matrix:
-  - PYTEST_VERSION=3.6.0 PLUGINCOMPAT_SITE=http://plugincompat.herokuapp.com
+  - PYTEST_VERSION=3.7.2 PLUGINCOMPAT_SITE=http://plugincompat.herokuapp.com
   global:
     secure: I7Lx3j8NZx29yHe2tgNyXQkbXVRDJDhmgYRH3N8Z9sYBOnhfuMAXYJhBn9xmlMFB6FzUs/+P5Ba0u9MO/P+bTqCz0fnkYH/3tC6NzOLxZHVtvnhgFftlUPgMNLmnwOKvpOaet7TpB1HYntjUGYR4watoqIKR6Wg2joTKgt3dlWE=
 script:
```

Comment:
Perhaps we can conditionally execute different scripts depending on the branch? If on `master`, execute `python run.py`, otherwise we execute `test_web.py`. What do you think?

Reply:
Sound good.  Actually executing `test_web.py` does some data injection thing, so it should be just executing `pytest`.  I'll have to look into conditional execution  in .travis.yml because haven't tried that before.

Reply:
Also, about the POST_TOKEN, can you think of a better way to manage that? Anyone can read it from the environment so  it's not a real secret, it may as well be plaintext...

Reply:
That variable is not set in pull requests, so only people with commit access would be able to get it.

Reply:
PRs don't get the decrypted `POST_TOKEN`, so the only way to access that would be for a malicious code to be merged into `plugincompat`, IIUC.

Reply:
> so it should be just executing pytest

Oh right, I meant `pytest test_web.py` 😁

Reply:
Ah! If I'm not mistaken, @wimglenn read the secret from the environment via a plugin which pytest-plugincompat then tests, not via a PR to this repo - correct? I didn't consider that one...

Couldn't `run.py` just remove `POST_KEY` from the environment after reading it (before the plugins are run) then?

Reply:
The problem is that a subprocess inherit parent environment, including variables the child process has no business knowing about. If you google you see it's accidentally [leaked](http://webcache.googleusercontent.com/search?q=cache:g0kJ6_DBPUYJ:plugincompat.herokuapp.com/output/pytest-figleaf-1.0%3Fpy%3Dpy36%26pytest%3D3.1.0+&cd=2&hl=en&ct=clnk&gl=us) other times [before](http://webcache.googleusercontent.com/search?q=cache:4QkMjaHtwZgJ:plugincompat.herokuapp.com/output/pytest-autochecklog-0.2.0%3Fpy%3Dpy27%26pytest%3D3.1.0+&cd=1&hl=en&ct=clnk&gl=us). You should assume the secret is public knowledge.

Could probably plug the leak up [here](https://github.com/pytest-dev/plugincompat/blob/3c7b9eb4d7ae3856a71e087ba579b10894ac0af8/run.py#L98-L99) - 
 you can use the `env` kwarg in subprocess call (also don't pass `shell=True`, which seems unnecessary here). However this is treating the symptoms not the cause, the best practice IMO is not using env vars for secrets because it's too easy to leak them accidentally (the env can get slured up by things like crash reporters, logging, etc).

Reply:
> The problem is that a subprocess inherit parent environment, including variables the child process has no business knowing about.

Hence why `run.py` should remove it from the environment after reading the value.

> However this is somewhat bandaid solution, the best practice IMO is not using env vars for secrets because it's too easy to leak them accidentally (the env can get slurped up by things like crash reporters, logging, etc).

It's pretty much the only way to get secrets into Travis CI, and also the one they support (by e.g. not making it available to PRs, not showing the value in the Travis logs, etc.).

> Split the current script which runs the results and uploads in two scripts:

Then the plugin running could just call `sudo` itself though.

Reply:
Secret changed and #27 opened with a tentative fix.

Reply:
Ah, well we have to work with the tools available. Popping it out of the environment and into the runtime seems good to me  👍